client/site-profiler/components/site-profiler-v2.tsx

```diff
@@ -89,26 +89,17 @@ export default function SiteProfilerV2( props: Props ) {
 
 	const url = useMemo( () => getValidUrl( routerDomain ), [ routerDomain ] );
 
-	const {
-		data: basicMetrics,
-		isFetching: isFetchingBasicMetrics,
-		error: errorBasicMetrics,
-	} = useUrlBasicMetricsQuery( url, hash, true );
-	const showBasicMetrics =
-		basicMetrics && basicMetrics.success && ! isFetchingBasicMetrics && ! errorBasicMetrics;
-
-	// TODO: Remove this debug statement once we have a better error handling mechanism
-	if ( errorBasicMetrics ) {
-		debug(
-			`Error fetching basic metrics for domain ${ domain }: ${ errorBasicMetrics.message }`,
-			errorBasicMetrics
-		);
-	}
+	const { data: basicMetrics } = useUrlBasicMetricsQuery( url, hash, true );
 
 	const showGetReportForm = !! url && isGetReportFormOpen;
 
 	const { data: performanceMetrics } = useUrlPerformanceMetricsQuery( routerDomain, hash );
 
+	const basicMetricsFromPerfReport = useMemo(
+		() => getBasicMetricsFromPerfReport( performanceMetrics ),
+		[ performanceMetrics ]
+	);
```

Comment:
I think it worth to check if react query sends a different object every time for `performanceMetrics` (I think that's the case).

In positive case, I would remove the `useMemo` as it is just increasing complexity as this value would be calculated at every render anyways.

Reply:
That's a very good point. I have checked, and the value returned is always the same in both scenarios, when there is no data, which is `undefined`, and when there is data. 

When there is data, the hook is called twice during the standard flow, but you can test opening the `get report form`, which is still called, and the object returned is always the same.